src/Dfe.PlanTech.Infrastructure.Redis/RedisCache.cs

```diff
@@ -253,16 +253,26 @@ private static CacheResult<T> CreateCacheResult<T>(RedisValue redisResult)
     }
 
     /// <inheritdoc/>
-    public Task InvalidateCacheAsync(string contentComponentId)
+    public Task InvalidateCacheAsync(string contentComponentId, string contentType)
     => _backgroundTaskService.QueueBackgroundWorkItemAsync(async (cancellationToken) =>
         {
             var key = _dependencyManager.GetDependencyKey(contentComponentId);
-            var dependencies = await GetSetMembersAsync(key);
-            foreach (var item in dependencies)
-            {
-                await RemoveAsync(item);
-            }
+            await RemoveDependenciesAsync(key);
 
-            await SetRemoveItemsAsync(key, dependencies);
+            // Invalidate all empty collections
+            await RemoveDependenciesAsync(_dependencyManager.EmptyCollectionDependencyKey);
```

Comment:
If `EmptyCollectionDependencyKey` is only used here, would there be any benefit to making a method on the `RedisDependencyManager` to handle this and the above instead?